src/components/questionnaire/questionnaires-edit-mode.tsx

```diff
@@ -0,0 +1,360 @@
+  };
+
+  /**
+   * Handle change event for answer option in question
+   *
+   * @param questionIndex number
+   * @param optionIndex number
+   * @param updatedFields - The updated fields for the question
+   */
+  const handleAnswerOptionChange = (
+    questionIndex: number,
+    optionIndex: number,
+    updatedFields: Partial<AnswerOption>
+  ) => {
+    const updatedOptions = editedQuestionnaire.questions[questionIndex].answerOptions.map(
+      (option, i) => (i === optionIndex ? { ...option, ...updatedFields } : option)
+    );
+
+    handleQuestionChange(questionIndex, { answerOptions: updatedOptions });
+  };
+
+  /**
+   * Functions to add new question to Questionnaire
+   *
+   * @param questionText string
+   * @param answerOptions - The answer options for the question
+   */
+  const handleAddQuestion = ({
+    questionText,
+    answerOptions
+  }: { questionText: string; answerOptions: AnswerOption[] }) => {
+    setEditedQuestionnaire((prevQuestionnaire) => ({
+      ...prevQuestionnaire,
+      questions: [...prevQuestionnaire.questions, { questionText, answerOptions }]
+    }));
+  };
+
+  /**
+   * Function to delete question from Questionnaire
+   *
+   * @param questionIndex number
+   */
+  const handleDeleteQuestion = (questionIndex: number) => {
+    const updatedQuestions = editedQuestionnaire.questions.filter(
+      (_, optionIndex) => optionIndex !== questionIndex
+    );
+    setEditedQuestionnaire((prev) => ({ ...prev, questions: updatedQuestions }));
+  };
+
+  /**
+   * Function to count all correct answers in the questionnaire, used for passScore determination
+   *
+   * @returns number of correct answers
+   */
+  const countEditedCorrectAnswers = () => {
+    return editedQuestionnaire.questions.reduce((count, question) => {
+      return count + (question.answerOptions?.filter((option) => option.isCorrect).length || 0);
+    }, 0);
+  };
+
+  const maxCorrectAnswers = countEditedCorrectAnswers();
+
+  /**
+   * Function to handle passScore change
+   *
+   * @param value number
+   */
+  const handlePassScoreChange = (value: number) => {
+    const maxCorrectAnswers = countEditedCorrectAnswers();
+    const passScore = Math.min(value, maxCorrectAnswers);
+    setEditedQuestionnaire((prev) => ({ ...prev, passScore }));
+  };
+
+  /**
+   * Function to update edited questionnaire
+   *
+   * @returns updated questionnaire
+   */
+  const updateEditedQuestionnaire = async () => {
+    setLoading(true);
+    try {
+      const updatedQuestionnaire = await questionnairesApi.updateQuestionnaires({
+        id: editedQuestionnaire.id as string,
+        questionnaire: {
+          ...editedQuestionnaire,
+          passedUsers: clearPassedUsers ? [] : editedQuestionnaire.passedUsers
+        }
+      });
+      setSnackbarOpen(true);
+      return updatedQuestionnaire;
+    } catch (error) {
+      setError(`${strings.error.questionnaireSaveFailed}, ${error}`);
```

Comment:
Is this error message specific to updating a questionnaire rather than just saving/ creating? It should be